src/org/zaproxy/zap/extension/brk/impl/http/BreakAddEditDialog.java

```diff
@@ -145,7 +145,10 @@ public String validateFields() {
 				return Constant.messages.getString("brk.brkpoint.error.regex"); 
 			}
 		}
-		return null;
+		if( (this.getStringValue(FIELD_STRING)).contains("#") && (Location.url.equals(this.strToLoc(this.getStringValue(FIELD_LOCATION)))) && (Match.contains.equals(this.strToMatch(this.getStringValue(FIELD_MATCH)))) ){
```

Comment:
Some of the parentheses could be removed (e.g. the ones surrounding `this.getStringValue(FIELD_STRING)` and following operands).